### ISSUE TITLE
'as' keyword in Dart makes problems

### ISSUE THREAD

**Issue by YazeedAlKhalaf** [dart-lang/mockito#633]
I am facing a problem running the `mockito` package to generate mocks when the file contains the as keyword. It shows dynamic instead of `ui.Image` from `dart:ui` that I imported in my file `as ui`.

Below is the issue reproducing code:
```
import 'dart:ui' as ui;

import 'package:flutter/material.dart';
import 'package:mockito/annotations.dart';

@GenerateMocks([SomeClass])
void main() {}

class SomeClass {
  void someMethod({
    required ui.Image image,
    Image? someImage,
  }) {
    print('someMethod');
  }
}
```

This is the generated mock:
```
// Mocks generated by Mockito 5.4.0 from annotations
// in app/test/mocks/mock_try.dart.
// Do not manually edit this file.

// ignore_for_file: no_leading_underscores_for_library_prefixes
import 'package:flutter/material.dart' as _i3;
import 'package:mockito/mockito.dart' as _i1;

import 'mock_try.dart' as _i2;

// ignore_for_file: type=lint
// ignore_for_file: avoid_redundant_argument_values
// ignore_for_file: avoid_setters_without_getters
// ignore_for_file: comment_references
// ignore_for_file: implementation_imports
// ignore_for_file: invalid_use_of_visible_for_testing_member
// ignore_for_file: prefer_const_constructors
// ignore_for_file: unnecessary_parenthesis
// ignore_for_file: camel_case_types
// ignore_for_file: subtype_of_sealed_class

/// A class which mocks [SomeClass].
///
/// See the documentation for Mockito's code generation for more information.
class MockSomeClass extends _i1.Mock implements _i2.SomeClass {
  MockSomeClass() {
    _i1.throwOnMissingStub(this);
  }

  @override
  void someMethod({
    required dynamic image, // LOOK HERE - this has dynamic type, should be Image from dart:ui
    _i3.Image? someImage,  // LOOK HERE - this works because the type is imported from material.dart
  }) =>
      super.noSuchMethod(
        Invocation.method(
          #someMethod,
          [],
          {
            #image: image,
            #someImage: someImage,
          },
        ),
        returnValueForMissingStub: null,
      );
}
```

**Comment by YazeedAlKhalaf** [comment]
If someone can tell me where the importing and adding `_i#` where `#` is a number happens I will be able to solve the issue.

**Comment by YazeedAlKhalaf** [comment]
no, it compiles and works fine.

`Image? someImage` is from `flutter/material.dart`

you can get the same problem if you just use the `required ui.Image image,` imported from `import 'dart:ui' as ui;` is used with the `as` keyword.

**Comment by srawlins** [comment]
Ah ok. The issue may be that analyzer doesn't know about `dart:ui`... not sure.

**Comment by yanok** [comment]
@YazeedAlKhalaf are you sure the issue is `as` keyword? Will it work if you just `import 'dart:ui';` without `as`?

Issue https://github.com/dart-lang/mockito/issues/662 looks related, but there is no `as`...

**Comment by yanok** [comment]
@YazeedAlKhalaf I assume you run `build_runner` with `dart run build_runner build`, please let me know if this is not the case. What are the outputs of `which flutter` and `which dart` on your machine? Are these binaries in the same directory? If not, could you try running `[path to flutter bin]/dart run build_runner build`? Does it fix the issue? Thanks!

**Comment by yanok** [comment]
Looking at the old issues found https://github.com/dart-lang/mockito/issues/579 that has the same problem. Turns out that's a bug in `package:build_resolvers` that was fixed in version 2.1.0.

@YazeedAlKhalaf  Could you please check `build_resolvers` version you are using and try to upgrade if it's below 2.1.0?

**Comment by yanok** [comment]
Closing, as there is no feedback and I'm pretty confident it's a manifestation of the `build_resolvers` issue.
